### ISSUE TITLE
Not enough memory

### ISSUE THREAD

**Issue by Turtlesfr** [yarnpkg/yarn#6414]
I'm trying to run a yarn install and get this error
`Error: ENOMEM: not enough memory, scandir '/home/toto/public_html/yo/akeneo/pim-community-standard'`
free -h gives me
`              total        used        free      shared  buff/cache   available
Mem:           251G        122G         12G         54G        117G         73G
Swap:           29G        7.4G         21G`

**Comment by Turtlesfr** [comment]
Here is the full error log
`Arguments: 
  /home/toto/bin/node /home/toto/.yarn/bin/yarn.js install

PATH: 
  /home/toto/.yarn/bin:/home/toto/.config/yarn/global/node_modules/.bin:/usr/local/cpanel/3rdparty/lib/path-bin:/usr/local/cpanel/3rdparty/lib/path-bin:/usr/local/cpanel/3rdparty/lib/path-bin:/usr/local/jdk/bin:/usr/kerberos/sbin:/usr/kerberos/bin:/usr/local/sbin:/usr/local/bin:/sbin:/bin:/usr/sbin:/usr/bin:/usr/X11R6/bin:/usr/local/bin:/usr/X11R6/bin:/root/bin:/opt/bin:/opt/cpanel/composer/bin:/home/toto/.local/bin:/home/toto/bin

Yarn version: 
  1.9.4

Node version: 
  6.11.2

Platform: 
  linux x64

Trace: 
  Error: ENOMEM: not enough memory, scandir '/home/toto/public_html/yo/akeneo/pim-community-standard'
      at Error (native)

npm manifest: 
  {
    "name": "pim-community-standard",
    "version": "2.0.0",
    "description": "The \"Akeneo Community Standard Edition\" distribution",
    "config": {
      "source": "vendor/akeneo/pim-community-dev",
      "check": "vendor/akeneo/pim-community-dev/webpack/check-dependencies.js"
    },
    "scripts": {
      "sync": "NODE_PATH=node_modules node $npm_package_config_check --source=$npm_package_config_source",
      "webpack": "yarn run sync && NODE_PATH=node_modules webpack --config $npm_package_config_source/webpack.config.js --env=prod",
      "webpack-dev": "yarn run sync && NODE_PATH=node_modules webpack --config $npm_package_config_source/webpack.config.js",
      "webpack-watch": "yarn run sync && NODE_PATH=node_modules webpack --progress --config $npm_package_config_source/webpack.config.js --watch"
    },
    "devDependencies": {
      "colors": "1.1.2",
      "dep-diff": "1.0.1",
      "read-pkg": "2.0.0"
    },
    "dependencies": {
      "JSON2": "0.1.0",
      "babel-cli": "6.26.0",
      "babel-core": "6.26.0",
      "babel-eslint": "7.2.3",
      "babel-loader": "7.1.2",
      "babel-minify": "0.2.0",
      "babel-polyfill": "6.26.0",
      "babel-preset-env": "1.6.0",
      "backbone": "0.9.10",
      "bundle-loader": "0.5.5",
      "codemirror": "5.25.2",
      "deepmerge": "1.5.1",
      "eslint": "4.5.0",
      "exports-loader": "0.6.4",
      "expose-loader": "0.7.3",
      "html-loader": "0.5.1",
      "imports-loader": "0.7.1",
      "jquery": "1.11.0",
      "loader-utils": "1.1.0",
      "lodash": "4.17.4",
      "merge-objects": "1.0.5",
      "raw-loader": "0.5.1",
      "regexp-replace-loader": "1.0.0",
      "summernote": "0.6.5",
      "text-loader": "0.0.1",
      "underscore": "1.8.3",
      "webpack": "3.5.5",
      "webpack-cleanup-plugin": "0.5.1",
      "webpack-dev-server": "2.7.1",
      "webpack-livereload-plugin": "0.11.0",
      "write-file-webpack-plugin": "4.1.0",
      "yamljs": "0.3.0"
    }
  }

yarn manifest: 
  No manifest

Lockfile: 
  No lockfile`

**Comment by probil** [comment]
@Turtlesfr It may be related to inodes usage.
Could you please provide output of the `df -i` command?

**Comment by Turtlesfr** [comment]
I'm afraid it won't be helpful :/
`df: cannot read table of mounted file systems: No such file or directory`
According to the hosting statistics, I've got plenty of space.
I'll contact the hosting company and may have to change. I'm on a semi-managed dedicated server.

**Comment by probil** [comment]
Yes but free inodes is not the same free space. It's Unix-fs-specific thing.
https://en.wikipedia.org/wiki/Inode
I was faced with a similar issue on my server a few years ago.
Applications threw an error that there is not enough space. But the half of the disk was free!
The problem was that one of application created a lot of small files (0.5-2kb) and didn't manage to clear them. I've made a simple script that removes those old files and that fixed the issue

I hope you will find a way to fix it quickly :)
